### PR TITLE
feat(db): add description to entities and plugin fields

### PR DESCRIPTION
**Pull request by Guaris**
XR: https://github.com/Kong/kong/pull/10150

This PR contains added descriptions for all OSS plugins and core entities.

### CODE REVIEW

**Comment by jschmid1** [comment]
Nice work @Guaris 

there is a syntax issue here

```
ERROR: error loading module 'kong.db.schema.typedefs' from file './kong/db/schema/typedefs.lua':
	./kong/db/schema/typedefs.lua:681: '}' expected (to close '{' at line 678) near 'required'
```

and there seems to be a general error with

```
error loading module 'kong.db.schema.entities.consumers' from file './kong/db/schema/entities/consumers.lua':
	./kong/db/schema/entities/consumers.lua:12: '<name>' expected near '"The unique username of the Consumer. You must send either this field or custom_id with the request."'
	```

**Comment by Guaris** [comment]
@jschmid1 I think I fixed them.

**Comment by jschmid1** [comment]
Aand another one:

```
ERROR: ./kong/db/schema/typedefs.lua:562: attempt to perform arithmetic on a string value
```

File: kong/db/schema/typedefs.lua [hunk 1]
```diff
@@ -528,10 +558,11 @@ typedefs.hosts = Schema.define {
       err = "invalid wildcard: must be placed at leftmost or rightmost label",
     },
     custom_validator = validate_host_with_wildcards,
-  }
+  },
+  definition = "An array of strings representing hosts. A valid host is a string containing one or more labels separated by periods, with at most one wildcard label ("*")"
```
Comment (jschmid1) [inline kong/db/schema/typedefs.lua:562]:
```suggestion
  definition = "An array of strings representing hosts. A valid host is a string containing one or more labels separated by periods, with at most one wildcard label ('*')"
```

Comment (jschmid1) [reply kong/db/schema/typedefs.lua:562]: this caused the ` attempt to perform arithmetic on a string value`

**Comment by jschmid1** [comment]
closed in favor of #10765

**Pull request closed without merging**